js/main.js

```diff
@@ -55,7 +66,8 @@ class App extends React.Component {
     sessionID: null,
     panes: {},
     focusedPaneID: null,
-    envID: ACTIVE_ENV,
+    envID: localStorage.getItem( 'envID' ) || null,
+    envIDs: JSON.parse(localStorage.getItem( 'envIDs' )) || {},
```

Comment:
This default value is not working and causing the whole visdom page to not load on a clean pull. I tried using the default value `[]` but that hasn't worked either. It appears the render method of `TreeSelect` in `renderEnvControls` is returning null or undefined, which is making React error out.

Can you ensure this is working on a fresh pull without the saved environment data in localstorage and with a blank environment? I won't be able to give an in-depth review on what remains until I can test it out for myself.

Reply:
do you have the web console error for this? I tried this in chrome and firefox and they both work for me when localstorage is cleared.

Also, I think the default value should be `{}`, not `[]`

Reply:
also, just to double check, you did a `yarn install` right? Want to make sure that the rc-tree-select package was added to node_modules

Reply:
I had not, though it isn't actually changing anything (just tried with a yarn install). Ideally we shouldn't need to have a user run the install step as we hadn't needed to in the past.

![screen shot 2018-03-01 at 11 01 52 am](https://user-images.githubusercontent.com/1276867/36856159-556627e6-1d43-11e8-99f7-989de99730ec.png)

We may be running into dependency issues here - how are you building and running visdom locally?

Reply:
I'm using yarn (as was suggested in the contributing to the UI section. But the yarn.lock should automatically update the dependencies right?

And for the downloaded js scripts, you definitely want to update because I updated to the next version of react. Did you run the download.sh script that I had? Do that for now and then we can add it into the python file. I'm sorry this is a pain :(

Reply:
btw, this is my first time using yarn/npm. So if I'm doing things incorrectly, please feel free to correct me

Reply:
Ah okay things are working now on after downloading your specific versions.

No worries about difficulty using yarn/npm - we primarily use them for the contributing side to make webpack and react work properly, but we try to leave usage of them out of the final distribution. In that way we're somewhat atypical. You'll be able to work this out by updating the version numbers listed in the `download_scripts` function, but at least for now things are working so I can do some testing.